controllers/component_build_controller.go

```diff
@@ -214,17 +217,18 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		log.Info("Starting Pipelines as Code provision for the Component")
 
 		var pacAnnotationValue string
-		done, err := r.ProvisionPaCForComponent(ctx, &component)
-		if !done {
-			// Need to retry
-			if err != nil {
+		var pacPersistentErrorMessage string
+		err := r.ProvisionPaCForComponent(ctx, &component)
+		if err != nil {
+			if boErr, ok := err.(*boerrors.BuildOpError); ok && boErr.IsPersistent() {
+				log.Error(err, "Pipelines as Code provision for the Component failed")
+				pacAnnotationValue = PaCProvisionErrorAnnotationValue
+				pacPersistentErrorMessage = boErr.ShortError()
+			} else {
+				// transient error, retry
 				log.Error(err, "Pipelines as Code provision transient error")
+				return ctrl.Result{}, err
 			}
-			return ctrl.Result{}, err
-		}
-		if err != nil {
-			pacAnnotationValue = "error"
-			log.Error(err, "Pipelines as Code provision for the Component failed")
 		} else {
 			pacAnnotationValue = PaCProvisionDoneAnnotationValue
```

Comment:
would be nice to remove PaCProvisionErrorDetailsAnnotationName in case that it's there. It would be confusing to see that it's done but there is error-message.